mlir-sycl/doc/design/SYCLKernelSubmissionRepresentation.md

```diff
@@ -0,0 +1,304 @@
+# Representation of SYCL kernel submission to scheduler
+
+## Introduction
+
+A SYCL command-group function (CGF) submits a kernel function to the runtime's
+scheduler to execute over a given range and with a given set of arguments.
+Capturing this contextual information is essential for host-device
+optimizations, such as constant propagation and simplification of accessors.
+
+In this document, we introduce the `sycl.host.schedule_kernel` operation, and
+describe its multi-stage raising process from LLVM-IR involving additional
+intermediate operations.
+
+The operations discussed in this document are defined in
+`mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLHostOps.td`. Build the target
+`mlir-sycl-doc` in a build directory named `build` to enable the links to the
+operations' rendered Markdown documentation.
+
+## Prerequisites / known limitations
+
+- Our approach is currently limited to kernels passed as a lambda function to an
+overload of `sycl::handler::parallel_for`.
+-  We rely on
+`sycl::handler::StoreLambda` being inlined into `sycl::handler::parallel_for`,
+and the latter being inlined into the CGF.
+- The auto-generated *rounded range kernel* should be deactivated with `-Xclang
+  -fsycl-disable-range-rounding`.
+
+## The `sycl.host.schedule_kernel` operation
+
+The
+[`schedule_kernel`](../../../build/docs/Dialects/SYCLOps.md#syclhostschedule_kernel-syclsyclhostschedulekernel)
```

Comment:
Can't we have a link that make sense when installed/deployed ? this is highly specific to a setup